core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -414,6 +415,46 @@ protected DiffRepository getDiffRepos() {
         .check();
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3171">[CALCITE-3171]
+   * SemiJoin on conditions push down throws IndexOutOfBoundsException</a>. */
+  @Test public void testPushSemiJoinConditionsToLeft() {
+    final RelBuilder relBuilder = RelBuilder.create(RelBuilderTest.config().build());
+    // build a rel equivalent to sql:
```

Comment:
minor errors in SQL description:
where emp.**deptno** 
in (select dept.deptno from dept where **emp**.empno **>** 20)

Reply:
Yep, thanks for correcting.

Reply:
Sorry for being picky, there are still typos :)
where emp.dempno => where emp.deptno 
where emp.empno < 20 => where emp.empno > 20

Reply:
Oops, i'm really careless these days.

Reply:
No problem, thanks for fixing it!